src/composed/message/types.rs

```diff
@@ -413,9 +413,8 @@ impl Message {
                 let msg = Message::from_bytes(data.decompress()?)?;
                 msg.verify(key)
             }
-            // Nothing to do for others.
-            // TODO: should this return an error?
-            _ => Ok(()),
+            // We don't know how to verify a signature for other Message types, and shouldn't return Ok
+            _ => Err(Error::Unsupported("Unexpected message format".to_string())),
```

Comment:
lets add the message type to the error

Reply:
Good point. I added `Debug` output for the `Message` (which might be long, but I couldn't find anything better to print?)

Reply:
good enough for now, though we should probably add a display impl for these that is a bit more usable

Reply:
(in a follow up/later I mean)

Reply:
sounds good